### PR TITLE
DellEMC: Z9332f platform API changes

### PR DESCRIPTION
**Pull request by aravindmani-1**
#### Why I did it
platform test suite failed for few API's in DellEMC Z9332f platform.
#### How I did it
Modified the API's to return the expected values in the script.
#### How to verify it
Run platform test suite after making the changes.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [x] 202106

#### Description for the changelog
Platform test suite has passed after making the API changes.
UT: 
[z9332f_UT.txt](https://github.com/Azure/sonic-buildimage/files/6877574/z9332f_UT.txt)

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py [hunk 1]
```diff
@@ -1178,7 +1191,7 @@ def get_tx_power(self):
                 if channel_monitor_data is not None:
                     tx1_pw = self._strip_unit_from_str(channel_monitor_data['data']['TXPower']['value'])
                 else:
-                    tx1_pw = 0.0
+                    tx1_pw = "N/A"
```
Comment (prgeor) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py:1194]: tx_power_list should be empty if the channel_monitor_data is empty due to eeprom read failure. By appending 'N/A' the test will pass even though the eeprom read failed.  The tx_power_list should be 'N/A' iff self.qsfp_dd_txpower_supported is false.

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py [hunk 2]
```diff
@@ -1129,7 +1138,7 @@ def get_rx_power(self):
                 if rx_power_data is not None:
                      rx1_pw = self._strip_unit_from_str(rx_power_data['data']['RXPower']['value'])
                 else:
-                     rx1_pw = 0.0
+                     rx1_pw = "N/A"
```
Comment (prgeor) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py:1141]: rx_power_data[] should be empty due to eeprom read failure. By appending 'N/A' the test will pass even though the eeprom read failed. The rx_power_list[] should be 'N/A' iff self.qsfp_dd_rxpower_supported is false.

**Review by prgeor — COMMENTED** [review]
is_xcvr_optical() needs fix for QSFP-DD https://github.com/Azure/sonic-mgmt/blob/5697eea34d4d5261ff69350098ce67188b8b1c81/tests/platform_tests/api/test_sfp.py#L150

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py (hunk 2 above)
Comment (aravindmani-1) [reply platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/sfp.py]: Done

**Comment by aravindmani-1** [comment]
> is_xcvr_optical() needs fix for QSFP-DD https://github.com/Azure/sonic-mgmt/blob/5697eea34d4d5261ff69350098ce67188b8b1c81/tests/platform_tests/api/test_sfp.py#L150

https://github.com/Azure/sonic-mgmt/pull/3888 has been raised to fix the test script issue.

**Review by prgeor — APPROVED** [review]

**Review by sujinmkang — APPROVED** [review]

**Pull request merged**